src/java.net.http/share/classes/jdk/internal/net/http/Http1Exchange.java

```diff
@@ -342,8 +349,11 @@ CompletableFuture<ExchangeImpl<T>> sendHeadersAsync() {
 
                         if (debug.on()) debug.log("requestAction.headers");
                         List<ByteBuffer> data = requestAction.headers();
-                        synchronized (lock) {
+                        lock.lock();
+                        try {
                             state = State.HEADERS;
```

Comment:
Can this code be replaced by a call to `switchState(...)` to be consistent with the rest of the places in this class which do a similar call?